### ISSUE TITLE
No source found for (eval ...)

### ISSUE THREAD

**Issue by gpetrowitsch** [Camelcade/Perl5-IDEA#2266]
I'm debugging a large Perl app. At one point, it evals code read in from a file. When I 'step into' the eval, the code (which naturally comes in a string) is shown in a new editor tab, even perfectly highlighted. However, as soon as I want to single-step through this eval'ed code it opens another tab which shows this text:

    # No source found for (eval 3369)[Perl-Lib/VeraCGWidget.pm:171]

This points to the location, where the eval command is.

If I let the program continue, it does so without problems, but I can't single step through the eval.

  * IDE name and version: PyCharm Community 2019.3 Build #PC-193.6494.30
  * Perl5 plugin version: 2019.3
  * JRE version: 1.8.0_201-b09
  * Your OS name and version: Windows 10 Enterprise

**Comment by hurricup** [comment]
Probably I msssing something, but I  need a minimal reproducible example. Just tried myself and it works fine. 

Test file
```
use strict;
use warnings FATAL => 'all';
use v5.10;

use Foo::Baz::Eval;
say 1;
Foo::Baz::Eval::declare;
Foo::Baz::Eval::foobar();
say 2;
```
`Foo::Baz::Eval.pm`
```
package Foo::Baz::Eval;
use strict;
use warnings FATAL => 'all';
use strict;
use v5.10;

sub declare{
    eval <<eval;
    say "evaled  in " . __PACKAGE__;
    sub foobar{
    say 42;
    say 43;
    say 44;
    }
eval
    say "Result is $@";
}
1;
```
Works and debugged as expected

**Comment by hurricup** [comment]
I've tried on Linux though, you may try my example on windows

**Comment by gpetrowitsch** [comment]
Thanks for your reply. I'm a bit short on time at the moment. Maybe it has something to do with what is in the eval code. I paste one of these snippets below.
It puts some surrounding stuff around the actual user code, which here is only the line with the "writeWord" call.
Part of this stuff puts the user code in a different name space, so it cannot easily mess with the internals of my application.
There are also some extra semicolons, which come in through the way, the code is added and to make it safe, if the user code doesn't have a trailing semicolon. The call "use VeraCG qw(:ALL)" just imports some variables and subroutines from the VeraCG package.

    my $Radio = $self->{radio};;
    my $Config = $self->{doConfig} ? 1 : 0; my $MySelf = $self->{ID};
    package VeraCG::M1_T1;
    use VeraCG qw(:ALL);

      $Lab{Florina}->writeWord(0x0100, $Radio<<0, 0x0003);
    ;
    package VeraCG;

    ;

**Comment by gpetrowitsch** [comment]
I forgot to mention, that the debugger steps in at the "use VeraCG..." line and the error occurs as soon as I do a single step over this line.

**Comment by hurricup** [comment]
Probably I wasn't clear. I don't need your real code examples. 

Please, when you have time, try to make a script (or scripts) that you can share and which demonstrates your problem. So I could run it locally, reproduce and fix. For now, I can't reproduce it.

**Comment by gpetrowitsch** [comment]
No, you were clear. But when it runs on your side, there must be something in my code, which causes the problem. But melting down my huge application to something small that still shows the problem will take time, which I don't have at the moment...

**Comment by hurricup** [comment]
For start you could use my snippets above and see if problem persists there
